pkg/transport/keepalive_listener.go

```diff
@@ -29,7 +29,12 @@ func NewKeepAliveListener(addr string, scheme string, info TLSInfo) (net.Listene
 		return nil, err
 	}
 
-	if scheme == "https" {
+	switch scheme {
+	case "http":
+		if !info.Empty() {
+			return nil, fmt.Errorf("cannot listen on non TLS for %s when TLS files are presented", scheme+"://"+addr)
```

Comment:
k. This stop behavior is too aggressive, and may lead regression problem. will change.